app/controllers/businesses_controller.rb

```diff
@@ -0,0 +1,72 @@
+class BusinessesController < ApplicationController
+  def index
+    if params[:search]
+      @businesses = Business.search(params[:search])
+    else
+      @businesses = Business.all
+    end
+  end
+
+  def new
+    @business = Business.new
+  end
+
+  def create
+    @business = Business.new(business_params)
+    @categories = params[:categories].split(",").map {|category| category.strip}
+
+    if @business.save
+      @categories.each do |skill|
+        @business.categories << Category.find_or_create_by(name: category)
+      end
+      if params[:images]
+        params[:images].each do |image|
+          @business.photos.create(image: image)
+        end
+      end
+      session[:business_id] = @business.id
+      redirect_to business_path(@business)
+    else
+      render :new
+    end
+  end
+
+  def show
+    @business = Business.find(params[:id])
+    @owner = @business.owner
+    @hash = Gmaps4rails.build_markers(@business) do |business, marker|
```

Comment:
Call this @map_data or similar (so that when you look at the view you know what it is.)